.travis.yml

```diff
@@ -1,8 +1,36 @@
+sudo: false
 language: node_js
-node_js:
-  - "6"
-before_script:
-  - npm install -g elm elm-test elm-doc-test
-  - elm-test init
-  - elm-doc-test
-script: elm-test tests/Doc/Main.elm
+node_js: "node"
+os: linux
+env: ELM_VERSION=0.18.0
+
+cache:
+  directories:
+    - test/elm-stuff/build-artifacts
+    - sysconfcpus
+
+  before_install:
+    - echo -e "Host github.com\n\tStrictHostKeyChecking no\n" >> ~/.ssh/config
+    - | # epic build time improvement - see https://github.com/elm-lang/elm-compiler/issues/1473#issuecomment-245704142
+      if [ ! -d sysconfcpus/bin ];
+      then
+      git clone https://github.com/obmarg/libsysconfcpus.git;
+      cd libsysconfcpus;
+      ./configure --prefix=$TRAVIS_BUILD_DIR/sysconfcpus;
+      make && make install;
+      cd ..;
+      fi
+
+  install:
+    - node --version
```

Comment:
Couldn't we stick to something simpler ? Like just installing the dependencies and running the tests.

Reply:
I would like to go with a simpler `.travis.yml` but from my experiments with elm a simple `.travis.yml` leads to very long test time or test that never complete. I think the other elm package repos have complicated `.travis.yml`'s for a reason.